tools/autograph-client/client.go

```diff
@@ -39,6 +39,25 @@ type signatureresponse struct {
 	X5U        string `json:"x5u,omitempty"`
 }
 
+type requestType int
+const (
+	_ = iota
+	requestTypeData
+	requestTypeHash
+	requestTypeFile
+)
+func urlToRequestType(url string) requestType {
+	if strings.HasSuffix(url, "/sign/data") {
+		return requestTypeData
+	} else if strings.HasSuffix(url, "/sign/hash") {
+		return requestTypeHash
+	} else if strings.HasSuffix(url, "/sign/file") {
+		return requestTypeFile
+	} else {
+		panic(fmt.Sprintf("Unrecognized request type for url", url))
```

Comment:
I prefer log.Fatal

Reply:
I'll switch this, but for testing I've been preferring panic since I can catch it.